packages/dialect-react-sdk/hooks/useNotificationChannel.ts

```diff
@@ -0,0 +1,197 @@
+  delete: () => Promise<void>;
+
+  verify: (params: VerifyParams) => Promise<void>;
+  resend: () => Promise<void>;
+
+  isCreatingAddress: boolean;
+  isUpdatingAddress: boolean;
+  isDeletingAddress: boolean;
+  isSendingCode: boolean;
+  isVerifyingCode: boolean;
+
+  isFetching: boolean;
+  errorFetching: DialectSdkError | null;
+}
+
+interface UseAddressesParams {
+  addressType: AddressType;
+  refreshInterval?: number;
+}
+
+function useGlobalAddress({
+  addressType,
+  refreshInterval,
+}: UseAddressesParams): UseAddressesValue {
+  const { wallet: walletsApi } = useDialectSdk();
+
+  const [isCreatingAddress, setCreatingAddress] = useState(false);
+  const [isUpdatingAddress, setUpdatingAddress] = useState(false);
+  const [isDeletingAddress, setDeletingAddress] = useState(false);
+  const [isVerifyingCode, setVerifyingCode] = useState(false);
+  const [isSendingCode, setSendingCode] = useState(false);
+
+  // Fetch all wallet addresses (it doesn't include a enabled prop)
+  const {
+    data: addresses = EMPTY_ARR,
+    error: errorFetchingAddresses = null,
+    mutate: mutateAddresses,
+  } = useSWR(
+    WALLET_ADDRESSES_CACHE_KEY_FN(walletsApi),
+    walletsApi ? () => walletsApi.addresses.findAll() : null,
+    {
+      refreshInterval,
+      refreshWhenOffline: true,
+    }
+  );
+
+  const isFetchingAddresses =
+    Boolean(walletsApi) && !errorFetchingAddresses && addresses === undefined;
+
+  const address = addresses.find((it) => it.type === addressType);
+
+  const errorFetching = errorFetchingAddresses;
+
+  const createAddress = useCallback(
+    async ({ value }: CreateParams) => {
+      if (isCreatingAddress) {
+        // Do not create address if it's already creating
+        return null;
+      }
+      setCreatingAddress(true);
+      try {
+        // not sure if optimistic update needed here, cause:
+        // - address creating is the action that user expect to take a while
+        await walletsApi.addresses.create({
+          type: addressType,
+          value,
+        });
+        return (
+          (await mutateAddresses())?.find((it) => it.type === addressType) ||
+          null
+        );
+      } finally {
+        setCreatingAddress(false);
+      }
+    },
+    [addressType, isCreatingAddress, mutateAddresses, walletsApi]
+  );
+
+  const updateAddress = useCallback(
+    async ({ value }: UpdateParams) => {
+      if (!address || isUpdatingAddress) {
+        return null;
+      }
+      setUpdatingAddress(true);
+      try {
+        await walletsApi.addresses.update({
+          addressId: address.id,
+          value,
+        });
+        return (
+          (await mutateAddresses())?.find((it) => it.type === addressType) ||
+          null
+        );
+      } finally {
+        setUpdatingAddress(false);
+      }
+    },
+    [
+      address,
+      addressType,
+      isUpdatingAddress,
+      mutateAddresses,
+      walletsApi.addresses,
```

Comment:
also create/delete using walletApi object. There we get specific addresses. What's the reason of this?

Reply:
eslint puts deps like this. I prefer to have whole walletApi, replaced!